src/components/graph/search/GraphSearch.tsx

```diff
@@ -75,6 +75,8 @@ function GraphSearch(props: Props) {
 	const wrapperRef = React.useRef<HTMLDivElement>(null);
 	const modalRef = React.useRef<HTMLDivElement>(null);
 
+	const [timeMode, setTimeMode] = React.useState(false);
```

Comment:
i think the better pattern to fetch event is to have submittedTimestamp: Number | null
then in GraphSearchDialog you can have effect

```
	React.useEffect(() => {
		const searchChannels = {
			previous: submittedTimestamp ? getEventSource(SearchDirection.Previous) : null,
			next: submittedTimestamp ? getEventSource(SearchDirection.Next) : null,
		};

		return () => {
			searchChannels.next?.stop();
			searchChannels.previous?.stop();
		};
	}, [submittedTimestamp]);

```
  in your current implementation you are searching not closest event, but first that you received, this might be a problem.

Also after event has been found this component makes 4 another http requests with ?probe=true for some reason

Reply:
Http request were sent because of fetchObjectById. There were no check for time and because of that were sent requests for found id and for inputted time.